### PR TITLE
improve travis build

### PR DESCRIPTION
**Pull request by return42**
In this series I will try to improve the travis build a little bit.

### CODE REVIEW

**Comment by return42** [comment]
Just to left some notes here .. 

I don't know why, but in the travis build there is no pip command in `./local/py3/bin`,  See _commands available at./local/py3/bin_ : https://travis-ci.org/github/asciimoo/searx/jobs/669664131#L146

Before changing `os: linux` in commit 8c145ff there was a pip .. strange.

To be able to do without it, commit 9b48ae4 does `replace: $(PY_ENV_BIN)/pip --> $(PY_ENV_BIN)/python -m pip`

**Comment by dalf** [comment]
For information, doesn't require answers, I'm not sure to understand:
* arch: arm64  # what does it bring ?
* before_cache: ...

**Comment by return42** [comment]
> before_cache: 

See commit message https://github.com/asciimoo/searx/pull/1910/commits/9e32fb27e71ca0ae32a0eead712a0504c9d2d951 .. I had the same "Permission denied: '/home/travis/.cache/pip/...' " issue in PR #1803 and solved it by a change owner in the before_cache phase of the travis build.  I don't know why the same solution does not work in this PR .. travis caching drives me a bit crazy.

**Comment by return42** [comment]
> arch: arm64 # what does it bring 

nothing, I can rewind it.

[EDIT] face palm .. I confused arm with amd .. n.c.

**Comment by return42** [comment]
Even if I [disable cache](https://travis-ci.org/github/asciimoo/searx/jobs/669685945/config?utm_medium=notification&utm_source=github_status), "Permission denied: '/home/travis/.cache/pip/...' "  error pops up. WTH is it.

**Comment by dalf** [comment]
About amd64, there is this error :
```/home/travis/.travis/functions: line 109: /home/travis/firefox-latest/firefox/firefox: cannot execute binary file: Exec format error```

So the tests won't be able to run successfully.

**Comment by return42** [comment]
Strange, now the github GUI hangs at "Some checks haven't completed yet" .. the build itself is done with success https://travis-ci.org/github/asciimoo/searx/pull_requests

**Comment by dalf** [comment]
I've restarted the build on travis. Now github synchronized the status.

**Comment by return42** [comment]
> I will also test the builds step by step, don't merge this until I am asking for / thanks!

@dalf what do you think .. is this PR ready to merge?

File: .travis.yml [hunk 1]
```diff
@@ -27,6 +31,7 @@ stages:
 jobs:
   include:
     - python: "2.7"
+      env: PY=2
```
Comment (dalf) [inline .travis.yml:32]: ~~Why / how this variable is used ?~~
to skip the test.pylint target with the Python 2 version.

File: .travis.yml [hunk 2]
```diff
@@ -5,7 +5,9 @@ language: python
 cache:
   - directories:
     - $HOME/.cache/pip
-
```
Comment (dalf) [inline .travis.yml:6]: From https://github.com/falconry/falcon/blob/master/.travis.yml#L78-L79 : 
> NOTE(vytas): wheels sometimes fail to build on s390x due to cache being owned by another user.

So, perhaps these two new lines can be removed on the amd64 architecture.

Comment (dalf) [reply .travis.yml:6]: It works without these two lines:
* https://travis-ci.org/github/dalf/searx/builds/670155742/config
* https://github.com/dalf/searx/blob/return42-rfc-travis/.travis.yml

File: .travis.yml (hunk 1 above)
Comment (return42) [reply .travis.yml:32]: You can run make targets with PY=2 to test in or install a python 2 environment:

      make PY=2  install test

If PY= is unset, PY=3 is the default.  To see more options run help

     make help

Comment (return42) [reply .travis.yml:32]: The PY=2 condition in main Makefile was only a needed addition to skip the pylint test. Mainly the PY variable sets a py2 or py3 environment.

File: .travis.yml (hunk 2 above)
Comment (return42) [reply .travis.yml:6]: Thanks a lot for the tested solution .. I applied 41f002b

**Review by dalf — APPROVED** [review]
Thank you for dealing with travis !

Note: perhaps squash the commits since there are some trails / errors / fixes.

**Comment by return42** [comment]
Let them in the history ..

**Pull request merged**